tests/chainer_tests/functions_tests/activation_tests/test_rrelu.py

```diff
@@ -3,82 +3,76 @@
 import numpy
 
 import chainer
-from chainer import backend
 from chainer import cuda
 from chainer import functions
-from chainer import gradient_check
 from chainer import testing
 from chainer.testing import attr
 
 
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'cuda_device': [0, 1],
+    })
+)
 @testing.parameterize(*testing.product({
     'train': [True, False],
-    'shape': [(3, 2), (5, 6)],
+    'shape': [(3, 2), ()],
     'dtype': [numpy.float16, numpy.float32, numpy.float64],
 }))
-class TestRReLU(unittest.TestCase):
+class TestRReLU(testing.FunctionTestCase):
+
+    dodge_nondifferentiable = True
 
     def setUp(self):
-        # Avoid unstability of numerical grad
-        self.x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
-        self.x[(-0.05 < self.x) & (self.x < 0.05)] = 0.5
-        self.gy = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
         # Assumption l < u
         self.l = numpy.random.uniform(0, 1)
         self.u = numpy.random.uniform(0, 1)
         if self.l >= self.u:
             self.l, self.u = self.u, self.l
-        self.check_forward_options = {}
-        self.check_backward_options = {}
+
         if self.dtype == numpy.float16:
             self.check_forward_options = {'atol': 1e-4, 'rtol': 1e-3}
             self.check_backward_options = {'atol': 5e-4, 'rtol': 5e-3}
+            self.check_double_backward_options = {'atol': 5e-4, 'rtol': 5e-3}
 
-    def check_forward(self, x_data):
-        x = chainer.Variable(x_data)
-        xp = backend.get_array_module(x)
-        with chainer.using_config('train', self.train):
-            y, r = functions.rrelu(x, l=self.l, u=self.u, return_r=True)
```

Comment:
It seems the test of `return_r` has been removed.

Reply:
- `return_r` is tested in `TestRReLUR`.
- There's no longer a test with `r=None, return_r=False`, (which are defaults!)  I'll add a test to confirm `F.rrelu` just runs with that options.

Reply:
I didn't notice it.
It still lacks a test for backward/double-backward for generated `r`s, but it's another issue.